### PR TITLE
add-support-for-irsa

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/roadmap/issues/114

### CODE REVIEW

File: service/internal/cloudconfig/tccpn.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ import (
 	"github.com/giantswarm/aws-operator/service/internal/hamaster"
 )
 
+const IRSAAnnotation = "alpha.aws.giantswarm.io/iam-roles-for-service-accounts"
+
```
Comment (calvix) [inline service/internal/cloudconfig/tccpn.go:29]: temporary until we have the annotation in `k8smetadata` repo

File: service/internal/cloudconfig/tccpn.go [hunk 2]
```diff
@@ -133,7 +135,7 @@ func (t *TCCPN) newTemplate(ctx context.Context, obj interface{}, mapping hamast
 		return "", microerror.Mask(err)
 	}
 
-	var cl infrastructurev1alpha3.AWSCluster
+	var awsCluster infrastructurev1alpha3.AWSCluster
```
Comment (calvix) [inline service/internal/cloudconfig/tccpn.go:138]: had to rename it to have a proper name for `cluster` cr

File: service/internal/cloudconfig/tccpn_extension.go [hunk 3]
```diff
@@ -255,7 +257,35 @@ func (e *TCCPNExtension) Files() ([]k8scloudconfig.FileAsset, error) {
...
+						Name: FileOwnerGroupName,
+					},
+					User: k8scloudconfig.User{
+						Name: FileOwnerUserName,
+					},
+				},
+				Permissions: 0644,
+			}
+
+			certsMeta = append(certsMeta, encryptionConfig, serviceAccountV2Pub, serviceAccountV2Priv)
```
Comment (calvix) [inline service/internal/cloudconfig/tccpn_extension.go:288]: add new 2 files for service account v2

File: service/internal/cloudconfig/tccpn.go [hunk 4]
```diff
@@ -268,17 +307,25 @@ func (t *TCCPN) newTemplate(ctx context.Context, obj interface{}, mapping hamast
...
+		if key.OIDCGroupsClaim(awsCluster) != "" {
+			apiExtraArgs = append(apiExtraArgs, fmt.Sprintf("--oidc-groups-claim=%s", key.OIDCGroupsClaim(awsCluster)))
+		}
+
+		// enable IRSA on the api
+		if _, ok := cluster.Annotations[IRSAAnnotation]; ok {
+			apiExtraArgs = append(apiExtraArgs, "--service-account-key-file=/etc/kubernetes/ssl/service-account-key-v2.pub")
+			apiExtraArgs = append(apiExtraArgs, "--service-account-signing-key-file=/etc/kubernetes/ssl/service-account-key-v2.pem")
+			apiExtraArgs = append(apiExtraArgs, fmt.Sprintf("--service-account-issuer=https://s3-%s.amazonaws.com/%s-%s-oidc-pod-identity", key.Region(awsCluster), cc.Status.TenantCluster.AWS.AccountID, key.ClusterID(&cr)))
+			apiExtraArgs = append(apiExtraArgs, "--api-audiences=sts.amazonaws.com")
```
Comment (calvix) [inline service/internal/cloudconfig/tccpn.go:328]: add arguments to api for IRSA

File: service/internal/cloudconfig/tccpn.go [hunk 5]
```diff
@@ -251,6 +273,23 @@ func (t *TCCPN) newTemplate(ctx context.Context, obj interface{}, mapping hamast
...
+				},
+				&secret)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			serviceAccountV2Pub = string(secret.Data[key.ServiceAccountV2Pub])
+			serviceAccountV2Priv = string(secret.Data[key.ServiceAccountV2Priv])
+
+			return nil
+		})
```
Comment (calvix) [inline service/internal/cloudconfig/tccpn.go:291]: fetch the new service account files

**Review by AverageMarcus — APPROVED** [review]

**Pull request merged**